### ISSUE TITLE
How to add additional search paths to `IguanaTex`?

### ISSUE THREAD

**Issue by mihaiconstantin** [Jonathan-LeRoux/IguanaTex#55]
I am trying to use `IguanaTex` in conjunction with `minted` that requires `pygmentize`.

Suppose I have the following `path.tex` document:

```tex
\documentclass{article}

\pagestyle{empty}

\begin{document}

\catcode`:=\active
\def:{\par}

\parindent0pt\tt

\input|"echo $PATH"

\bigskip

Is there pygmentize?

\input|"which pygmentize"

\end{document}
```

Running this document from my editor (i.e., or the terminal as `pdftex --shell-escape path.tex`) produces:

![terminal](https://github.com/Jonathan-LeRoux/IguanaTex/assets/20051042/04ea6d0a-e185-4b45-8d56-9a5a9f544328)

However, running the same code from within `IguanaTex` yields:

![IguanaTex](https://github.com/Jonathan-LeRoux/IguanaTex/assets/20051042/99718fdf-29a0-4bf6-9e8c-aa6567430f4f)

My question for you is two-fold:

1. I would like to be able to add `/opt/homebrew/bin` or `/usr/local/bin` to the paths `IguanaTex` can see, but I cannot see a way to do this via the `Main Settings` tab. Is there currently a way to achieve this?

2. Is there a way I can instruct `IguanaTex` on what flags to use? For example, can I use the `--shell-escape` flag?

P.S. Thanks a lot for this amazing piece of software!

**Comment by mihaiconstantin** [comment]
My current workaround is to create a symbolic link to `pygmentize` in the `/Library/TeX/texbin` folder as follows:

```bash
ln -s /opt/homebrew/bin/pygmentize pygmentize
```

Then, running the `path.tex` in `IguanaTex` produces:

![IguanaTex](https://github.com/Jonathan-LeRoux/IguanaTex/assets/20051042/1a3d4e0d-4b9f-4d5c-aa7e-3d0c0c547271)

The code is now properly formatted through `minted.` However, I think it could still be handy to have an option to allow `IguanaTex` to see *non-standard* user-specified paths.

**Comment by Jonathan-LeRoux** [comment]
Thanks for reporting this and for finding a workaround. 

To answer question 2 first, `--shell-escape` is already used in the LaTeX calls.

I don't know why @tsung-ju coded it this way, maybe to circumvent the Mac sandbox, but there is a path export in [ShellWait.bas](https://github.com/Jonathan-LeRoux/IguanaTex/blob/22ffd80795d55d136bc788b412fc172e723af445/ShellWait.bas#L12) that is currently set to be the same path as that of the LaTeX executables. One solution would be to offer in the Main Settings yet another path that could be included in there as well. Do you think that would be enough? One way to check this would be for you to edit the .pptx to hard-code the path to `pygmentize` and see if that works.

**Comment by Jonathan-LeRoux** [comment]
I've added the possibility to specify additional paths in the Main Settings on Mac. They're added to the PATH export when executing any command. Multiple paths can be specify by separating with ":".
If I add `/opt/homebrew/bin` as additional path, then I get the correct behavior you describe, while it doesn't work if I don't add it.
Please give it a shot: [IguanaTex_v1_61_beta4.zip](https://github.com/Jonathan-LeRoux/IguanaTex/files/14661530/IguanaTex_v1_61_beta4.zip)

**Comment by mihaiconstantin** [comment]
Thanks for making this change @Jonathan-LeRoux! Can you please include the `.ppam` in the archive so I can try it out on my Mac?

**Comment by mihaiconstantin** [comment]
It seems to work well—no need for creating symbolic links to external tools anymore! Thank you for adding this option.

<img width="271" alt="image" src="https://github.com/Jonathan-LeRoux/IguanaTex/assets/20051042/438a27e1-7591-48d5-9cc1-61e27fecc70e">

---

> Note that you can easily get the .ppam from the .pptm by using "export" (on Mac, "save as" on Windows) and choose .ppam as the file format.

Ah, I wasn't aware of this... useful to know!

**Comment by Jonathan-LeRoux** [comment]
Closing this issue as it is now fixed in [Release v1.61](https://github.com/Jonathan-LeRoux/IguanaTex/releases/tag/v1.61).
